src/main/java/net/minecraftforge/client/resource/SensitiveReloadStateHandler.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.client.resource;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.Stack;
+
+import com.google.common.collect.Sets;
+
+/**
+ * Handles reload parameters for sensitive loaders.
+ */
+public enum SensitiveReloadStateHandler
+{
+    INSTANCE;
+
+    public static final ReloadRequirements ALLOW_ALL = new ReloadRequirements.Exclude(Collections.emptySet());
+
+    private final Stack<ReloadRequirements> currentRequirements = new Stack<>();
```

Comment:
A stack is very old and outdated. I suggest using an ArrayDeque and Deque interface instead.